go/vt/schemamanager/schemaswap/schema_swap.go

```diff
@@ -211,48 +202,52 @@ func (schemaSwap *Swap) stopAllHealthWatchers() {
 // the the method just picks up that. Otherwise it starts a new one and writes into the database that
 // the process was started.
 func (schemaSwap *Swap) initializeSwap(sql string) error {
-	metadataChannel := make(chan *shardSwapMetadata)
-	for _, shard := range schemaSwap.allShards {
-		go shard.readShardMetadata(metadataChannel)
-	}
-	var initError error
-	var cntDataReceived int
-	for cntDataReceived != len(schemaSwap.allShards) {
-		metadata := <-metadataChannel
-		cntDataReceived++
-		if metadata.lastStartedSwap == metadata.lastFinishedSwap {
+	var waitGroup sync.WaitGroup
+	metadataList := make([]shardSwapMetadata, len(schemaSwap.allShards))
+	for i, shard := range schemaSwap.allShards {
+		waitGroup.Add(1)
+		go shard.readShardMetadata(&metadataList[i], &waitGroup)
```

Comment:
Yes, I agree it's more readable, and I've tried exactly this already. :) I've got these errors on commit:

go/vt/schemamanager/schemaswap/schema_swap.go:211: range variable shard captured by func literal
go/vt/schemamanager/schemaswap/schema_swap.go:211: range variable i captured by func literal

Please fix the go vet warnings above. To disable certain checks, change vetflags in misc/git/hooks/govet.